admin_app/src/components/ChatComponents.tsx

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import Markdown from "react-markdown";
+import TextAreaAutosize from "react-textarea-autosize";
+
+import { backendUrl } from "../components/Config";
+
+export type Message = {
+  type: "human" | "ai";
+  content: string;
+};
+
+interface SubmitMessageProps {
+  setMessages: React.Dispatch<React.SetStateAction<Message[]>>;
+  messages: Message[];
+}
+export const SubmitMessage: React.FC<SubmitMessageProps> = ({
+  setMessages,
+  messages,
+}) => {
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const form = e.target as HTMLFormElement;
+    const formData = new FormData(form);
+
+    const formJson = Object.fromEntries(formData.entries());
+    console.log(formJson.query_text.toString());
+
+    const queryMessage: Message = {
+      type: "human",
+      content: formJson.query_text.toString(),
+    };
+
+    setMessages([...messages, queryMessage]);
+
+    // pop-up to get token if no token present
+    let token: string | null = localStorage.getItem("apiToken");
```

Comment:
This is a very hacky solution. Something I'll fix in the next PR

Reply:
Yeah! Problems are: 
* there is no way to edit your token once you have entered it
* so if you enter is wrong, you may have to restart your browser 🙈 

There is the cosmetic aspect of it but also the functional. I'll fix it in the next PR.